### PR TITLE
Port cholesky to structured kernels

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #79300

Yeah.

### CODE REVIEW

**Comment by Lezcano** [comment]
@IvanYashchuk PTAL

File: aten/src/ATen/native/BatchLinearAlgebra.cpp [hunk 1]
```diff
@@ -1853,132 +1871,44 @@ Tensor& cholesky_out(const Tensor &self, bool upper, Tensor &result) {
...
+TORCH_IMPL_FUNC(linalg_cholesky_ex_out)(const Tensor& A,
+                                        bool upper,
+                                        bool check_errors,
+                                        const Tensor& L,
+                                        const Tensor& info) {
+  // Nothing to do there
+  if (L.numel() == 0) {
+    info.zero_();
+    return;
   }
```
Comment (IvanYashchuk) [inline aten/src/ATen/native/BatchLinearAlgebra.cpp:1883]: Does anything fail without this special-casing?

**Review by IvanYashchuk — APPROVED** [review]

File: aten/src/ATen/native/BatchLinearAlgebra.cpp (hunk 1 above)
Comment (Lezcano) [reply aten/src/ATen/native/BatchLinearAlgebra.cpp:1883]: IIRC there was something that was breaking, but I don't remember what.

**Review by albanD — APPROVED** [review]
Nice clenaup!

**Pull request closed without merging**